src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1881,22 +1881,32 @@ namespace Opm {
         return { tmp.begin(), tmp.end() };
     }
 
-    std::vector< Well* > Schedule::getWells(const std::string& wellNamePattern) {
-        size_t wildcard_pos = wellNamePattern.find("*");
+    std::vector< Well* > Schedule::getWells(const std::string& wellNamePattern, const std::vector<std::string>& matching_wells) {
+        // If we arrive here during the handling the body of a ACTIONX keyword
+        // we can arrive with wellname '?' and a set of matching wells.
+        if (wellNamePattern == "?") {
+            std::vector<Well*> wells;
+            for (const auto& well_name : matching_wells)
+                wells.push_back( std::addressof(m_wells.get(well_name) ));
 
-        if( wildcard_pos != wellNamePattern.length()-1 ) {
-            if( !m_wells.hasKey( wellNamePattern ) ) return {};
-            return { std::addressof( m_wells.get( wellNamePattern ) ) };
-        }
+            return wells;
+        } else {
+            size_t wildcard_pos = wellNamePattern.find("*");
```

Comment:
Technically, `std::string::find()` returns `std::string::size_type`.  There's no guarantee that that will fit into a `size_t`, so this would be better expressed as
```C++
const auto wildcard_pos = /* ... */
```
More worrisome however, is that `std::string::find()` returns `std::string::npos` if the pattern does not exist, so the subsequent check
```C++
wildcard_pos != wellNamePattern.length() - 1
```
will be `true` if the `*` character does not exist (for normally sized `wellNamePattern` strings).

Reply:
> There's no guarantee that that will fit into a `size_t`

You might be formally right - and I have changed to use `auto` - which is anyway a good thing; but it would be a cold day in h...  - and  a highly non-standard string implementation - when the length of a string could not fit in a `std::size_t`.

Reply:
> will be true if the * character does not exist (for normally sized wellNamePattern strings).

yes - and that is actually OK. The current code has two states: 

1. Wellname pattern which *ends* with a `*` - that is passed to a matching algorithm.
2. Anything else - we look for the verbatim well name - with or without a `*` as part of the name.

So - I think this OK. That being said support for the `WLIST` keyword requires more elaborate matching here, but that is for another day.